internal/controllers/cluster/cluster_controller.go

```diff
@@ -331,6 +332,7 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	}
 
 	controllerutil.RemoveFinalizer(cluster, clusterv1.ClusterFinalizer)
+	r.recorder.Eventf(cluster, corev1.EventTypeNormal, "Deleted", "Cluster %s has been deleted", cluster.Name)
```

Comment:
is this a common pattern? Wondering if pods do this too, and/or if there isn't some kind of general Object deleted event that gets emitted

Reply:
For a pod, there's the `Killing` Event that's emitted when it is deleted. Here's the lifecycle events of creating a statefulset and then running a kubectl delete against the pod.

```
LAST SEEN   TYPE      REASON                            OBJECT                                            MESSAGE
60s         Normal    Scheduled                         pod/touge-debug-0                                 Successfully assigned default/touge-debug-0 to ip-10-158-59-46.us-west-2.compute.internal
4s          Normal    SuccessfulCreate                  statefulset/touge-debug                           create Pod touge-debug-0 in StatefulSet touge-debug successful
59s         Normal    Pulling                           pod/touge-debug-0                                 Pulling image "mtougeron/touge-debug"
46s         Normal    Pulled                            pod/touge-debug-0                                 Successfully pulled image "mtougeron/touge-debug" in 13.117780542s
46s         Normal    Created                           pod/touge-debug-0                                 Created container ubuntu
46s         Normal    Started                           pod/touge-debug-0                                 Started container ubuntu
3s          Normal    Killing                           pod/touge-debug-0                                 Stopping container ubuntu
4s          Normal    Pulled                            pod/touge-debug-0                                 Container image "mtougeron/touge-debug" already present on machine
4s          Normal    Scheduled                         pod/touge-debug-0                                 Successfully assigned default/touge-debug-0 to ip-10-158-59-46.us-west-2.compute.internal
3s          Normal    Created                           pod/touge-debug-0                                 Created container ubuntu
3s          Normal    Started                           pod/touge-debug-0                                 Started container ubuntu
```